js/java.js

```diff
@@ -175,3 +175,16 @@ function openProject() {
 
 iconClose.forEach((n) => n.addEventListener('click', closePopup));
 projectLink.forEach((n) => n.addEventListener('click', openProject));
+
+/** validate email address */
+const form = document.getElementById('form');
+form.addEventListener('submit', (e) => {
+  const email = document.getElementById('email');
+  const error = document.getElementById('error');
+  const pattern = /^([a-zd.-_]+)@([a-zd_-]+).([a-z-_]{2,8})(.[a-z-_]{2,8})?$/;
```

Comment:
- [ ] Kindly make sure that the validation works whenever there's an upper case character in the email. Currently, if the email contained an upper case character, it shows that it's valid.

- [Hint] (In regex, you can use `-` to specify a range of characters. Like what you have on `a-z`, specifying all the lower case characters. On the other hand, you also have `.-_` which means that you're selecting the range of characters between `.` and `_` which happens to also have all the upper case characters. You can check this [here](https://www.rapidtables.com/code/text/ascii-table.html#:~:text=46,_). To fix this all you have to do is to write `._-` instead of `.-_`)